test/config/utility.cc

```diff
@@ -0,0 +1,96 @@
+#include "test/config/utility.h"
+
+#include "common/common/assert.h"
+#include "common/protobuf/utility.h"
+
+#include "test/test_common/environment.h"
+#include "test/test_common/network_utility.h"
+
+#include "api/filter/http_connection_manager.pb.h"
+
+namespace Envoy {
+
+std::string basic_config = R"EOF(
+static_resources:
+  listeners:
+    name: listener_0
+    address:
+      socket_address:
+        address: 127.0.0.1
+        port_value: 0
+    filter_chains:
+      filters:
+        name: http_connection_manager
+        config:
+          httpFilters:
+            name: router
+            config:
+              deprecated_v1: true
```

Comment:
I haven't been fully following but why is this necessary?

Reply:
We just need to add createFilterFactoryFromProto to https://github.com/lyft/envoy/blob/master/source/server/config/http/router.h.

Reply:
I see. Shouldn't we do that anyway for all of the internal filters? Or at least the ones we are using in v2 integration tests?

Reply:
Or is the plan to have v2 filter config still use v1 JSON in existing cases?

Reply:
Added a tracking issue at https://github.com/lyft/envoy/issues/1619. For router, it's trivial, for others, e.g. Mongo, it will involve a bit of work. Definitely something to mop up.